### ISSUE TITLE
Exploration: moving options

### ISSUE THREAD

**Issue by karmatosed** [WordPress/gutenberg#19128]
The toolbar does a lot right now, the right side specifically is a quite intense combination of different interaction types.

![image](https://user-images.githubusercontent.com/253067/70806811-09906c00-1db4-11ea-9f66-4640da07b42e.png)

- Yellow: status
- Blue: publishing actions
- Red: sidebar / settings
- Green: options

It makes sense to me to explore reducing this. For now,  I wanted to start by examining where options could go.

Reducing back to positions, the following graphic shows how they fall right now:

<img width="684" alt="1" src="https://user-images.githubusercontent.com/253067/70807066-9d623800-1db4-11ea-97c5-8dec3772b8f8.png">

- Top left: block library
- Right: forward-moving actions such as publishing
- Bottom right: a proposed new place for options

The new position I wanted to investigate draws on other apps such as Figma, which uses a drawer for keyboard interactions.

A little note, the work explored here does bring in some elements from #18667 to see what could be. Specifically, the pop-menu uses the style from that Figma file. 

<img width="1238" alt="menu-options" src="https://user-images.githubusercontent.com/253067/70808272-385c1180-1db7-11ea-846c-56351a561423.png">
This moves options to the bottom and has the same icon. It could be great to explore what icon works here. I think some options could be:

![image](https://user-images.githubusercontent.com/253067/70808181-0f3b8100-1db7-11ea-947d-041a77bdabd8.png)

<img width="1237" alt="menu-closed" src="https://user-images.githubusercontent.com/253067/70808046-c4ba0480-1db6-11ea-8967-003bc7a482c9.png">

<img width="1238" alt="menuv2" src="https://user-images.githubusercontent.com/253067/70808226-1ebaca00-1db7-11ea-9524-c8348f4a6313.png">

Another option is a smaller strip.

<img width="1238" alt="menu-view" src="https://user-images.githubusercontent.com/253067/70808238-27ab9b80-1db7-11ea-9661-081487e6531a.png">

Other panels could be:

<img width="1238" alt="menu-options" src="https://user-images.githubusercontent.com/253067/70808289-4316a680-1db7-11ea-8083-7f52bf531c45.png">

I explored a full panel that is open, it could reduce back to a menu and modal that we have, however, I think to grow this and be more useful progressing past the modal could be worth exploring, particularly as this opens to extensibility.

<img width="1245" alt="menu-pop" src="https://user-images.githubusercontent.com/253067/70808312-4ad64b00-1db7-11ea-85bc-01434d4aa4fe.png">

What do people think? This is very early-stage sketching out the idea for now so would welcome some feedback.

**Comment by shaunandrews** [comment]
I think moving editor-related settings to the bottom-right makes a lot of sense. I'd push to continue exploring with a menu, rather than a full pane with tabs and the like — the pane is just way too much UI for what should be a really simple thing.

I think the ellipsis icon should likely still live at the top-right though, and should be focused on plugins which add their own sidebar — think of it like an overflow menu for icons that open sidebars.

With that said, I think a cog or wrench icon could work for the editor settings on the bottom-right. And we could change the current inspector sidebar icon to an ( i ) info icon, removing the current document outline icon.

Here's a visual explanation of the above:

<img width="1323" alt="image" src="https://user-images.githubusercontent.com/191598/70808997-eacbae80-1d8e-11ea-8020-2377a0548827.png">

**Comment by shaunandrews** [comment]
Changing the current document/block inspector icon might be _too_ much. I explored using a different icon (wrench) and a simple text label for the bottom-right editor settings as well:

<img width="618" alt="image" src="https://user-images.githubusercontent.com/191598/70809478-d63be600-1d8f-11ea-8d65-5b5843160d1a.png">

**Comment by karmatosed** [comment]
A variation then could be to rather than have modal take this approach:

<img width="1245" alt="suggest" src="https://user-images.githubusercontent.com/253067/70811341-5593de80-1dbd-11ea-8333-b3376dcdae13.png">

**Comment by melchoyce** [comment]
> Changing the current document/block inspector icon might be too much. I explored using a different icon (wrench) and a simple text label for the bottom-right editor settings as well:

Combining these two could be good — we have so many icons that essentially all kind of mean the same thing (settings cog, settings wrench, more ••• ... all kind of the same idea, more stuff in a menu) that reinforcing with a label would go far to improve clarity.

**Comment by enriquesanchez** [comment]
>This is very early-stage sketching out the idea for now so would welcome some feedback.

Loving these early explorations @karmatosed! 

I agree with @shaunandrews, I feel like the tabbed pane feels a bit too much and think that the menu exploration might be worth pursuing. I wonder if we can even take inspiration from list menu patterns found in mobile and the Customizer?

As for accessibility feedback, I understand these are early explorations and nothing is set in stone yet, but it's also a great tine to think about topics such as document structure and keyboard navigation. We might find that having the options placed at the bottom right corner might prove to be hard to find and get to for some users.

**Comment by karmatosed** [comment]
@enriquesanchez is there anything we can do about order to help that? I know it's been said before being at top might also be a problem because forcing an order and being in area of publishing flow.

**Comment by enriquesanchez** [comment]
@karmatosed What I do is to try to imagine I'm using a screen reader and then ask myself "how would I know this element exists and how would I get to it?".

<img width="621" alt="Screen Shot 2019-12-13 at 17 40 10" src="https://user-images.githubusercontent.com/3276087/70838901-f390a480-1dcf-11ea-805f-a7d70c125389.png">

As you can see from the Landmarks selection modal in VoiceOver, we have a few editor-related landmarks or regions in Gutenberg. Where would I go if I wanted to get to this Options menu? My natural reaction would be to try the 'Editor top bar region' or 'Editor settings' region.

It would be frustrating and tedious to go to any of them, tab through all the options and realize it's not there. And this is assuming I know this menu even exists.

And to clarify, it's not that 'Editor footer region' would be the wrong place for this menu, it's just that given all the regions we already have—and how they're named—that it gets confusing to figure out where would it be.

I do think that the simplification and logical grouping you're exploring on this issue is important  and if we take a holistic view at this and how we can make use of regions and grouping to bring more logic to the interface, it has the potential to make navigating around the editor easier.

**Comment by chrisvanpatten** [comment]
One other element that I want to toss into the ring is that the number of options in that top right area expands greatly as you get into very complicated Gutenberg implementations.

For example, this is our top right UI:

<img width="502" alt="Screen Shot 2019-12-14 at 9 36 25 AM" src="https://user-images.githubusercontent.com/1231306/70850160-36e12680-1e55-11ea-843d-7b5b1e5761f2.png">

Each of these icons is for a sidebar panel which offers important features to our editorial and production teams. We're trying to do our best to expose the sidebars only when necessary, but ultimately we have to balance having fewer sidebars that do more and don't tell cohesive stories (e.g. mixing unrelated options) and having _many_ sidebars, which are each clear in their purpose.

While our case is maybe on the more extreme end, as we have many, many stakeholders across our brands interacting with the CMS and each group has competing needs, I'm sure we aren't the only ones with a UI like this. It would be nice to see explorations take these cases into account. Implementers such as us have very few options for exposing "document metadata settings" in the UI, and it would be nice to have a better way to visualize the available options such that they aren't buried in a menu or you get icon soup like our current approach.

(Happy to give a tour of our implementation as well, if it would ever be helpful!)

**Comment by jasmussen** [comment]
The challenge Chris shows here is worth thinking more about. Currently any sidebar can be unpinned (click the star), which _should_ mean that the sidebar only exists in the overflow menu on the right. 

For example, install Yoast and you can open the sidebar from the button, or the item in the ellipsis menu. Unpin it, and it can _only_ be opened from the item in the ellipsis menu.

If we remove/move the ellipsis menu, how does this interface work? It was inspired by how Chrome extensions work:

![extensions](https://user-images.githubusercontent.com/1204802/70902340-d42a8f00-1ffc-11ea-81e0-5f61bb37ce48.gif)

It arguably hasn't been intuitive or as successful as we hoped, but we still need to address how sidebar pinning works. Unless a user can explicitly opt out of these buttons some day we'll end up in a very bad UI place.

**Comment by rickybanister** [comment]
The points from Chris and Joen resonate with me. I was thinking about Slack's information architecture the other day. All of their 'top bar' icons open and populate their sidebar when clicked, and the ellipsis is a menu of less important sidebar content. It feels right that if the ellipsis sticks around it would have the same behavior, providing access to sidebar content. Display options/editor settings could simply be another sidebar content panel.

**Comment by shaunandrews** [comment]
> which should mean that the sidebar only exists in the overflow menu on the right.

What if we take this further and and keep the ellipsis menu _only_ for plugins:

![sidebar icons take 3](https://user-images.githubusercontent.com/191598/72645982-f2104a00-3942-11ea-96c0-f4409da62a5a.gif)

For people without any plugins, the menu would be reduced down:

<img width="940" alt="image" src="https://user-images.githubusercontent.com/191598/72646380-0274f480-3944-11ea-9168-61e4eb46ea62.png">

**Comment by karmatosed** [comment]
> For people without any plugins, the menu would be reduced down:

What about using '...' or an arrow to denote more plugins?

**Comment by jasmussen** [comment]
Keep the explorations going! I do worry about a single ellipsis button splitting into two, though — that feels like more UI and complexity, going against the initial goal.

**Comment by chrisvanpatten** [comment]
Instead of an ellipsis button, perhaps a down chevron similar to the button to view all RichText formats, would be more appropriate?

One thing I really dig about @shaunandrews' mockup is that it effectively promotes plugins to being standalone first class citizens, with their own interface.

Also as an update to [this comment](https://github.com/WordPress/gutenberg/issues/19128#issuecomment-565723737), we've since finished converting our "legacy" metaboxes to Gutenberg sidebars, and are now confronting this:

<img width="745" alt="Screen Shot 2020-03-02 at 4 44 04 PM" src="https://user-images.githubusercontent.com/1231306/75987190-7ef47000-5ebd-11ea-9f9a-35a788af6bd8.png">

(This is obviously a worse case scenario, partly due to our enterprise scale and the amount of metadata we need to catalog for every piece of content we publish.)

We have the luxury of being able to devote resources to iterating on that now that the initial conversion process is done (consolidating sidebars, figuring out where we can conditionally show/hide fields, thinking about what can be placed in pre/post publish panels, etc) but many users won't, because the sidebars are registered in third party plugins they can't control.

As such it might be nice to think about this alongside #20336, #14457, and similar "plugin experience" issues.

**Comment by ZebulanStanphill** [comment]
@chrisvanpatten
>Instead of an ellipsis button, perhaps a down chevron similar to the button to view all RichText formats, would be more appropriate?

Firefox has something pretty similar to this:
![image](https://user-images.githubusercontent.com/19592990/75996136-0a243480-5ec3-11ea-989f-127c2e38a031.png)

**Comment by richtabor** [comment]
Re the ellipsis more menu, we should get clear on what are options/settings vs. what should be high priority actions within the More menu. For example, the "views" at the top of the more menu are options - not actions. Whereas the remaining items are actions.

**Comment by paaljoachim** [comment]
We discussed this issue during a Gutenberg Design Triage:
https://wordpress.slack.com/archives/C02S78ZAL/p1594742741070300

An idea that come up was having an Interaction Design review going through the various main areas. We also need to think about how Full Site Editing will also affect the general UI. A lot of ideas came up such as removing the X in the sidebar. Moving the ellipse into the sidebar. Removing the Document and Block Tab. To just show the Block title when a Block is selected and the Document when no block is selected. (Would likely be a discoverability issue.)

Check the above link for the discussion that we had.

**Comment by afercia** [comment]
Any new position of the options should be considered not only visually but also as position in the DOM order. Please consider that for keyboard users and assistive technology users content navigation is a _linearized_ experience.

That said, the current placement of the options isn't ideal as well. Any improvement welcome :) 

This also brings us back to conversations made in a very early stage of this project, related to general organization of the main sections, especially the "Publish" panel in relation to the other sections.  In a logical, _linearized_, flow, the ideal order would be:
- post title
- content area: add blocks, edit blocks, etc.
- after user created their content: options
- publish/save/schedule

This mental model appears to be the most reasonable one and ideally it should be reflected in the UI.

Adding the Accessibility label as general layout of the main section in the block editor is one of the main features that affect an accessible user experience.

**Comment by afercia** [comment]
One more things to consider:

At the bottom of the block editor page there's already the meta boxes area. This area appears only when plugins add their own meta boxes in the "legacy" way. Any new design should take into consideration the bottom part of the page is already "taken" by the meta boxes area.

**Comment by karmatosed** [comment]
For now as this is my ticket originally and idea I am going to close this because there has been a lot of work around the design system that makes some of the original thinking not aligned. This is great and work can focus in that direction now. Thanks, everyone.
